src/EnergyPlus/WaterUse.cc

```diff
@@ -1193,16 +1108,109 @@ namespace WaterUse {
+                        // print error for variables of hot water temperature
+                        ++this->CWHWTempErrorCount;
+                        TempDiff = this->ColdTemp - this->HotTemp;
+                        if (this->CWHWTempErrorCount < 2) {
+                            ShowWarningError(state,
+                                             format("CalcEquipmentFlowRates: \"{}\" - Hot water temperature is less than the cold water "
+                                                    "temperature by ({:.2R} C)",
+                                                    this->Name,
+                                                    TempDiff));
+                            ShowContinueErrorTimeStamp(state, "");
+                            ShowContinueError(state, format("...hot water temperature        = {:.2R} C", this->HotTemp));
+                            ShowContinueError(state, format("...cold water temperature       = {:.2R} C", this->ColdTemp));
+                            ShowContinueError(state,
+                                              "...Hot water temperature should be greater than or equal to the cold water temperature. "
+                                              "Verify temperature setpoints and schedules.");
+                        } else {
+                            ShowRecurringWarningErrorAtEnd(
+                                state,
+                                format("\"{}\" - Hot water temperature should be greater than the cold water temperature error continues... ",
+                                       this->Name),
+                                this->CWHWTempErrIndex,
+                                TempDiff,
+                                TempDiff);
+                        }
+                    }
+                } else {
+                    // HotMassFlowRate should always be between 0 and TotalMassFlowRate
+                    this->HotMassFlowRate = this->TotalMassFlowRate * (this->TargetTemp - this->ColdTemp) / (this->HotTemp - this->ColdTemp);
+                }
             }
 
             this->ColdMassFlowRate = this->TotalMassFlowRate - this->HotMassFlowRate;
-
-            if (this->ColdMassFlowRate < 0.0) this->ColdMassFlowRate = 0.0;
-
             this->MixedTemp = (this->ColdMassFlowRate * this->ColdTemp + this->HotMassFlowRate * this->HotTemp) / this->TotalMassFlowRate;
+            // there should be no out of bounds results
+            assert(this->ColdMassFlowRate >= 0.0 && this->ColdMassFlowRate <= this->TotalMassFlowRate);
+            assert(this->HotMassFlowRate >= 0.0 && this->HotMassFlowRate <= this->TotalMassFlowRate);
+            assert(std::abs(this->HotMassFlowRate + this->ColdMassFlowRate - this->TotalMassFlowRate) < EPSILON);
         } else {
             this->HotMassFlowRate = 0.0;
-            this->ColdMassFlowRate = 0.0;
+            this->ColdMassFlowRate = this->TotalMassFlowRate;
             this->MixedTemp = this->TargetTemp;
```

Comment:
Now that the line above sets `this->ColdMassFlowRate = this->TotalMassFlowRate;` this line should be 
`this->MixedTemp = this->ColdTemp`

Reply:
I agree with you but lines 1040 - 1057 dictate the target so a wrong target with a water use object that has no connections and no hot water temp or target schedules should come out with whatever the user incorrectly entered. Based on the logic at line 590. If both schedules are missing they will get the right answer. I guess a warning if one or the other schedule is entered that it won't be used and then fix that specific target temp logic.

Reply:
Ok, I think I understand. You've examined this logic much more than I have, so Ok.